test/dlc_tests/ec2/test_smdebug.py

```diff
@@ -23,21 +23,22 @@ def test_smdebug_gpu(training, ec2_connection, region, gpu_only, py3_only):
     test_script = SMDEBUG_SCRIPT
     framework = get_framework_from_image_uri(training)
     container_test_local_dir = os.path.join("$HOME", "container_tests")
-    ec2_connection.run(
-        f"$(aws ecr get-login --no-include-email --region {region})", hide=True
-    )
+    ec2_connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
 
     ec2_connection.run(
         f"nvidia-docker run --name smdebug-gpu -v "
         f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} -itd {training}",
         hide=True,
     )
 
-    ec2_connection.run(
-        f"nvidia-docker exec --user root smdebug-gpu /bin/bash -c '{test_script} {framework}'",
-        hide=True,
+    test_output = ec2_connection.run(
+        f"nvidia-docker exec --user root smdebug-gpu /bin/bash -c '{test_script} {framework}'", hide=True, warn=True
     )
 
+    # If there is a failure running SMDebug, dump the whole logs instead of relying on fabric's logs, as those
+    # can get truncated
+    assert test_output.ok, f"SMDebug tests failed. Output:\n{test_output.stdout}"
```

Comment:
Is this a temporary change or a long term change ?
I will try the same approach for the flaky test.